### ISSUE TITLE
cannot read property 'offsetHeight' of undefined

### ISSUE THREAD

**Issue by davidmaxwaterman** [PolymerElements/iron-list#468]
The code here :

https://github.com/PolymerElements/iron-list/blob/master/iron-list.html#L1263

```
      this._iterateItems(function(pidx, vidx) {
        oldPhysicalSize += this._physicalSizes[pidx] || 0;
        this._physicalSizes[pidx] = this._physicalItems[pidx].offsetHeight;
        newPhysicalSize += this._physicalSizes[pidx];
        this._physicalAverageCount += this._physicalSizes[pidx] ? 1 : 0;
      }, itemSet);
```

the assignment to `oldPhysicalSize` checks if `this._physicalSizes[pidx]` is defined before assignment, yet the following line assumes the corresponding `offsetHeight` property does exist. If it doesn't exist, as in my case, this results in the error:

```
cannot read property 'offsetHeight' of undefined
```

Examining _physicalSizes and _physicalItems, it seems like they contain only a small subset of my app's `items`...indexed 0..22. However, the value of `pidx` is `4390`. As a matter of interest, `vidx`, which is never used in that function, has value `8737`). Note that iron-list's `items` list is `4391`, so I'm curious where `8737` comes from.

The stack trace when I block on this line shows that it originates from a `this.set` on a property of one of the items in `items`.

If anyone has any ideas how I can resolve this problem, please let me know.

**Comment by davidmaxwaterman** [comment]
I've managed to come up with a minimal example:

https://jsbin.com/kividi/edit?html,console,output

Instructions:

1. run it.
2. wait for list to appear
3. scroll to the bottom of the list
4. click on the last button (so it toggles from 'on' to 'off')

observe error in console.

**Comment by AlbertoFdzM** [comment]
I've got the same error.

This one is triggered when the `iron-list` hast more items than it could render (IE you have a 500 items array but you'r rendering only 20) and you update some of the subproperties of one of the visible array items using the polymer elements `set` function.

the `iron-list` function in charge to calculate the corresponding height of the changed item (`_updateMetrics`) is not getting the physical index correctly, in fact it is taking the original array index of the element.

So for example:
1. `iron-list` is rendering in screen the indexes 40 to 60 from an array `items` of length 500.
1. Execute `myElement.set('items.43.date', Date.now())`
1. Polymers notifies to `iron-list` that the path `items.43.date` has been updated
1. The height calculation function takes that 43 and try to find that index in its `physicalItems` array that goes from index 0 to 19 (without any relation with the previous index) and without any previous check it tries to get its `offsetHeight`
1. the physicalItem related with the `items.43` could be in any position of the array (from 0 to 19) due that `iron-list` repositions the DOM elements (recycling the HTML) giving cases in which the last visual element in the list is index 13 and the first visible item is index 14, second is index 15 an so on

- `Polymer` v1.9.2
- `iron-list` v2.0.4
